src/NLog.Extensions.Logging/Extensions/ConfigureExtensions.cs

```diff
@@ -61,6 +65,32 @@ public static ILoggingBuilder AddNLog(this ILoggingBuilder factory, NLogProvider
             }
             return factory;
         }
+
+        /// <summary>
+        /// Enable and configure NLog as a logging provider for buildable generic host (.NET Core 2.1+). Can be used in discrete containers as well. 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns>IHostBuilder for chaining</returns>
+        public static IHostBuilder UseNLog(this IHostBuilder builder)
+        {
+            if (builder == null) throw new ArgumentException($"{nameof(builder)} is null.");
+
+            builder.ConfigureServices(services =>
+            {
+                ConfigurationItemFactory.Default.RegisterItemsFromAssembly(
+                    typeof(ConfigureExtensions).GetTypeInfo().Assembly);
+
+                LogManager.AddHiddenAssembly(typeof(ConfigureExtensions).GetTypeInfo().Assembly);
+
+                services.AddSingleton(new LoggerFactory().AddNLog(new NLogProviderOptions
```

Comment:
I think it would be nice to pass the `NLogProviderOptions` optional to `UseNLog`

Reply:
Agreed